src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,346 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="DeltaSetOfT.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Common;
+using Org.OData.Core.V1;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an <see cref="IDeltaSet"/> that is a collection of <see cref="IDeltaSetItem"/>s.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix")]
+    [NonValidatingParameterBinding]
+    public class DeltaSet<TStructuralType> : Collection<IDeltaSetItem>, IDeltaSet where TStructuralType : class
+    {        
+        private Type _clrType;
+        IList<string> _keys;
+        NavigationPath _navigationPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeltaSet{TStructuralType}"/> class.
+        /// </summary>
+        /// <param name="keys">List of key names for the type</param>
+        public DeltaSet(IList<string> keys)           
+        {
+            _keys = keys;            
+            _clrType = typeof(TStructuralType);
+            _navigationPath = new NavigationPath(_clrType.Name, null);
```

Comment:
>_clrType [](http://example.com/codeflow?start=49&length=8)

Is this right?  This will be the type name, not the name of the navigation path, right?